TLM/TLM/Manager/Impl/SpeedLimitManagerExt.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace TrafficManager.Manager.Impl {
+    /// <summary>Adds extra helper functions to NetSegment and NetInfo.Lane, etc.</summary>
+    public static class SpeedLimitManagerExt {
+        /// <summary>
+        /// Extension method. Call via 'segment.MayHaveCustomSpeedLimits()`.
+        /// Check whether custom speed limits may be assigned to the given segment.
+        /// </summary>
+        /// <param name="segment">Reference to affected segment.</param>
+        /// <returns>True if this segment type can have custom speed limits set.</returns>
+        public static bool MayHaveCustomSpeedLimits(this ref NetSegment segment) {
+            if ((segment.m_flags & NetSegment.Flags.Created) == NetSegment.Flags.None) {
```

Comment:
Also need to bail if `NetSegment.Flags.Collapsed` is set (road destroyed by a disaster; it is now a pile of rubble).

Reply:
Collapsed roads can be bulldozed and rebuilt (the speed limit value will be lost), or they can be upgraded in place (then it should be retained). Shall i just throw in that extra flag and it will work?

Reply:
I was today years old when I learnt you could just upgrade collapsed roads instead of bulldoze+rebuild. 😮 

Do we retain the speed when roads are upgraded?

Reply:
Technically, the value will not be lost on collapse, so it should be retained. I do not own the disasters DLC, the thought of ruining my carefully crafted cities creates internal conflict in me 😂